### DISCUSSION TITLE
rolling file with size limits and auto rotation?

### DISCUSSION THREAD

**Discussion by keepsimple1** [tokio-rs/tracing#1726] [Q&A] [answered]
Hi, I am exploring to move from `log` to `tracing`.  It seems like `tracing-appender` crate has `RollingFileAppender` that supports rotation [based on a time period](https://docs.rs/tracing-appender/0.2.0/tracing_appender/rolling/struct.Rotation.html).  Does it (or some other crate) support rolling files based on a size limit and total-number-of-files limit?

i.e. something like this:  log tracing events into a file `foo.log` until it exceeds a given size limit, then automatically rename the current `foo.log` into `foo.log.1`, and then later `foo.log.2`, until a given limit of total number of log files.

When the total number of files limit is reached,  will start overwriting `foo.log.1` again.  (`foo.log` is always the newest log)

**Comment by hawkw** [comment, marked as answer]
This library might provide what you're looking for: https://crates.io/crates/rolling-file
